plugins/linkattacher/src/linkattacherserverimpl.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2013-2015 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include <linkattacherserverimpl.h>
+
+using namespace gazebo;
+using namespace std;
+using namespace yarp::os;
+
+LinkAttacherServerImpl::LinkAttacherServerImpl()
+{}
+
+LinkAttacherServerImpl::~LinkAttacherServerImpl()
+{}
+
+bool LinkAttacherServerImpl::attachUnscoped(const string& model_name, const std::string& model_link_name, const std::string& robot_name, const string& robot_link_name)
+{
+
+    #if GAZEBO_MAJOR_VERSION >= 8
+    gazebo::physics::ModelPtr object_model = _world->ModelByName(model_name);
+    #else
+    gazebo::physics::ModelPtr object_model = _world->GetModel(model_name);
+    #endif
+
+    if(!object_model)
+    {
+        yError() << LogPrefix << "Attach: " << model_name << " does not exist in gazebo";
+        return false;
+    }
+    else yInfo() << LogPrefix << "Attach: " << object_model->GetName() << " found";
+
+    gazebo::physics::LinkPtr object_link = object_model->GetLink(model_link_name);
+    if(!object_link)
+    {
+        yError() << LogPrefix << "Attach: " << model_link_name << " is not found";
+    }
+    else yInfo() << LogPrefix << "Attach: " << object_link->GetName() << " found";
+
+    #if GAZEBO_MAJOR_VERSION >= 8
+    gazebo::physics::ModelPtr robot_model = _world->ModelByName(robot_name);
+    #else
+    gazebo::physics::ModelPtr robot_model = _world->GetModel(robot_name);
+    #endif
+
+    if(!robot_model)
+    {
+        yError() << LogPrefix << "Attach: " << robot_name << " does not exist in gazebo";
+        return false;
+    }
+    else yInfo() << LogPrefix << "Attach: " << robot_model->GetName() << " found";
+
+
+    //Get the exact link with only link name instead of full_scoped_link_name
+    gazebo::physics::Link_V robot_model_links = robot_model->GetLinks();
+    for(int i=0; i < robot_model_links.size(); i++)
+    {
+        //E.g iCub::l_hand::l_hand_base_link
+
+        std::string candidate_robot_link_name = robot_model_links[i]->GetScopedName();
+
+        //Display all the links
+        //yInfo() << "Attach --> Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
+
+        std::size_t lastcolon = candidate_robot_link_name.rfind(":");
```

Comment:
Would it work also searching `::`. It looks that the separator is `::` and not just `:`

Reply:
You can also use the `hasEnding()` from `libraries/common` 

For example, please see lines `223-235`  [here](https://github.com/robotology/gazebo-yarp-plugins/blob/01ad1d1252b0827033fd57058f066cc790b893c6/plugins/contactloadcellarray/src/ContactLoadCellArrayDriver.cpp#L223)